x/signal/keeper.go

```diff
@@ -18,6 +20,11 @@ var (
 
 	// defaultSignalThreshold is 5/6 or approximately 83.33%
 	defaultSignalThreshold = sdk.NewDec(5).Quo(sdk.NewDec(6))
+
+	// defaultUpgradeHeightDelay is the number of blocks after a quorum has been
+	// reached that the chain should upgrade to the new version. Assuming a
+	// block interval of 12 seconds, this is 48 hours.
+	defaultUpgradeHeightDelay = int64(3 * 7 * 24 * 60 * 60 / 12) // 3 weeks * 7 days * 24 hours * 60 minutes * 60 seconds / 12 seconds per block = 151,200 blocks.
```

Comment:
yea can to make it const and public. 

We can discuss the duration of 3 weeks in today's v2 call. Yaz suggested 3 weeks so I went with that.

Reply:
Ok, I'll just state my reason here for posterity. Having a three week delay means all networks, include our testnets will take at minimum three weeks. This could mean that it may take two to three months from the initial final release to being on mainnet. Including auditing and other internal testing, this is frankly way too slow of a turnaround time. Especially when we want to try increase the cadence of upgrades (it's almost been 9 months since mainnet launch).

Reply:
We could make the delay period different durations based on the network. 
- Arabica: 1 week
- Mocha: 2 weeks
- Mainnet: 3 weeks

Reply:
Discussed synchronously and landed on 1 week.